src/env.cpp

```diff
@@ -160,6 +160,24 @@ NAN_METHOD(EnvWrap::open) {
     }
 }
 
+NAN_METHOD(EnvWrap::resize) {
+    Nan::HandleScope scope;
+
+    // Get the wrapper
+    EnvWrap *ew = Nan::ObjectWrap::Unwrap<EnvWrap>(info.This());
+
+    if (!ew->env) {
+        return Nan::ThrowError("The environment is already closed.");
+    }
+
```

Comment:
Actually, the node-lmdb `env` tracks its transactions (see `cleanupStrayTxns()`) so you could add some code here to enforce that no transactions are active.

Reply:
Something like this:

```cpp
if (env->currentWriteTxn || env->readTxns.size()) {
    return Nan::ThrowError("Only call env.resize() when there are no active transactions. Please close all transactions before calling env.resize()");
}
```